src/spec/unit/reader.spec.coffee

```
@@ -0,0 +1,150 @@
+  new Reader
+    csvDelimiter: delimiter,
+    encoding: encoding,
+    importFormat: fileType,
+
+describe 'Reader', ->
+  before ->
+    tempDir = tmp.dirSync({ unsafeCleanup: true })
+    Promise.resolve()
+
+  after ->
+    tempDir.removeCallback()
+    Promise.resolve()
+
+  it 'should read xlsx file', ->
+    tempFile = path.join(tempDir.name, 'data.xlsx')
+
+    writeXlsx(tempFile, dataInput)
+    .then ->
+      createReader('xlsx').read(tempFile)
+    .then (data) ->
+      expect(data).to.be.an('array')
+      expect(data.length).to.equal(1)
+      expect(data[0]).to.deep.equal(expectedOutput)
+
+  it 'should read csv file without provided options', ->
+    tempFile = path.join(tempDir.name, 'data-simple.csv')
+
+    writeCsv(tempFile, dataInput)
+    .then ->
+      (new Reader()).read(tempFile)
+    .then (data) ->
+      expect(data).to.be.an('array')
+      expect(data.length).to.equal(1)
+      expect(data[0]).to.deep.equal(expectedOutput)
+
+  it 'should read csv file with custom delimiter', ->
+    tempFile = path.join(tempDir.name, 'data.csv')
+
+    writeCsv(tempFile, dataInput, ';')
+    .then ->
+      createReader('csv', ';').read(tempFile)
+    .then (data) ->
+
+      expect(data).to.be.an('array')
+      expect(data.length).to.equal(1)
+      expect(data[0]).to.deep.equal(expectedOutput)
+
+  it 'should read csv file with custom encoding', ->
+    tempFile = path.join(tempDir.name, 'data-encoding.csv')
+
+    row = unicodeString + ',123'
+
+    dataInput = [
+      ['encodedString', 'numb']
+      iconv.encode(row, 'win1250')
+    ]
+
+    writeCsv(tempFile, dataInput)
+    .then ->
+      createReader('csv', undefined, 'win1250').read(tempFile)
+    .then (data) ->
+      expect(data).to.be.an('array')
+      expect(data.length).to.equal(1)
+      expect(data[0].encodedString).to.equal(unicodeString)
+      expect(data[0].numb).to.equal('123')
+
+
+  it 'should not read csv file with unsupported encoding', ->
```

You can return a promise as a result of the function and `mocha` will check if the promise ends with an error. This test tries to create an instance of Reader with unsupported encoding which should throw an error.